sdk/synapse/Azure.Analytics.Synapse.AccessControl/src/SynapseAccessControlClient.cs

```diff
@@ -54,6 +55,497 @@ public virtual Response<SynapseRoleAssignment> CreateRoleAssignment(SynapseRoleS
+            scope.Start();
+            try
+            {
+                var response = await GetRoleDefinitionByIdAsync(
+                    roleDefinitionName.ToString(),
+                    options: new RequestOptions()
+                    {
+                        CancellationToken = cancellationToken
+                    }).ConfigureAwait(false);
+
+                JsonDocument roleDefinitionJson = await JsonDocument.ParseAsync(response.ContentStream, default, cancellationToken).ConfigureAwait(false);
+                List<SynapsePermission> permissions = new List<SynapsePermission>();
+                foreach (var permissionsItem in roleDefinitionJson.RootElement.GetProperty("permissions").EnumerateArray())
+                {
+                    List<string> actions = new List<string>();
+                    List<string> notActions = new List<string>();
+                    List<string> dataActions = new List<string>();
+                    List<string> notDataActions = new List<string>();
+                    foreach (var actionsItem in permissionsItem.GetProperty("actions").EnumerateArray())
+                    {
+                        actions.Add(actionsItem.GetString());
+                    }
+                    foreach (var notActionsItem in permissionsItem.GetProperty("notActions").EnumerateArray())
+                    {
+                        notActions.Add(notActionsItem.GetString());
+                    }
+                    foreach (var dataActionsItem in permissionsItem.GetProperty("dataActions").EnumerateArray())
+                    {
+                        dataActions.Add(dataActionsItem.GetString());
+                    }
+                    foreach (var notDataActionsItem in permissionsItem.GetProperty("notDataActions").EnumerateArray())
+                    {
+                        notDataActions.Add(notDataActionsItem.GetString());
+                    }
+                    permissions.Add(new SynapsePermission(actions, notActions, dataActions, notDataActions));
+                }
+                List<SynapseRoleScope> scopes = new List<SynapseRoleScope>();
+                foreach (var scopesItem in roleDefinitionJson.RootElement.GetProperty("scopes").EnumerateArray())
+                {
+                    scopes.Add(scopesItem.GetString());
+                }
+
+                return Response.FromValue(new SynapseRoleDefinition(
+                    roleDefinitionJson.RootElement.GetProperty("id").ToString(),
+                    roleDefinitionJson.RootElement.GetProperty("name").ToString(),
+                    roleDefinitionJson.RootElement.GetProperty("description").ToString(),
+                    permissions,
+                    scopes),
+                    response);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        // TODO: change return type from list to pageable
```

Comment:
>      // TODO: change return type from list to pageable [](http://example.com/codeflow?start=2&length=55)

Have you got a chance to test the method after changing the returning type to pagable?

Reply:
No, according to our offline discuss, may be we should first add "x-ms-pageable" in swagger file, and then test the rewrite method based on the new generated LLC APIs?